src/sinks/statsd.rs

```diff
@@ -166,14 +169,30 @@ impl SinkConfig for StatsdSinkConfig {
     }
 }
 
+// Note that if multi-valued tags are present, this encoding may change the order from the input
+// event, since the tags with multiple values may not have been grouped together.
+// This is not an issue, but noting as it may be an observed behavior.
 fn encode_tags(tags: &MetricTags) -> String {
     let parts: Vec<_> = tags
-        .iter_single()
-        .map(|(name, value)| {
-            if value == "true" {
-                name.to_string()
-            } else {
-                format!("{}:{}", name, value)
+        .iter_sets()
+        .map(|(name, tag_value_set)| match tag_value_set {
+            TagValueSet::Empty => name.to_owned(),
+            TagValueSet::Single(tag_value) => match tag_value {
+                TagValue::Bare => name.to_owned(),
+                TagValue::Value(value) => format!("{}:{}", name, value),
+            },
+            TagValueSet::Set(index_set) => {
+                let parts: Vec<_> = index_set
+                    .iter()
+                    .map(|tag_value| match tag_value {
+                        TagValue::Bare => name.to_owned(),
+                        TagValue::Value(value) => {
+                            format!("{}:{}", name, value)
+                        }
+                    })
+                    .collect();
+
+                parts.join(",")
```

Comment:
What is the motivation for doing this long-hand versus using `tags.iter_all()`?

Reply:
No motivation, just lack of awareness. I'll see about updating to use that.

Reply:
Yeah ... that is a big improvement :]